src/runtime_src/xocl/core/device.cpp

```diff
@@ -713,7 +713,7 @@ get_cu_memidx() const
         mask &= cu->get_memidx_intersect();
 
       // select first common memory bank index if any
-      for (size_t idx=0; idx<mask.size(); ++idx) {
+      for (size_t idx=mask.size() - 1; idx>=0; --idx) {
```

Comment:
Please elaborate on this change.  The comment no longer reflects the loop.  Why was it changed?

Reply:
The OCL code should be using GROUP_CONNECTIVITY and the connectivity in this section should always be valid groups.

@saifuddin-xilinx explains that APPs requested backwards compatibility to old mem topology indices, but we have to revisit this request.  Grouping is a new feature, existing host code and xclbins will continue to work without any changes.  

However changing the xclbin to use grouping of HBMs and changing CU connectivity implies that host code must change also.   One should not expect existing host code to work with xclbins that have been converted to use groups.

We should resolve this and revert the change in xocl/core/device.cpp.